### DISCUSSION TITLE
Offer a Lua file for OSD customization similar to the one for OSC

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#15150] [Ideas]
MPV provides `osc.lua` file for users to customize the OSC.
From what I have been told there are no current ways to make modifications to the default OSD items that MPV provides other then some minor adjustments.
I am suggesting maybe a similar file can be provided `osd.lua` that would allow user to customize how these OSD items are displayed.

Some of the items that this osd.lua would contain are....
Volume OSD Bar
Duration OSD Bar

not as important but can also allow for parameters such as....
Brightness Bar
Saturation Bar
etc.

This osd.lua file would allow the user to edit things such as

`TEXT AT TOP`
-change what the default text displays
-move text at the top to a different location?
-add text items
-font / font size
-font colors

`ICONS`
-remove/ replace the Duration Play/Pause icon
-remove/ replace the Volume speaker icon
-placement
-size
-color

`BARS`
-bar colors
-fill the background in a color (requested here as an option #15149 but can also be added to this osd.lua file)
-move bar location (I have seen that this is already possible but can still be added to the osd.lua file)

Not sure if any developers come into the Discussions section?
Would any developers be interested in adding something like this?

Thank You

**Comment by guidocella** [comment]
osc.lua is not provided to customize the OSC, it's where it's implemented. To modify the OSD you have to modify the C code that implements it or add options.

**Comment by advice2020** [comment]
Well yes from what I understand, it is compiled into MPV, but, MPV allows user to copy that osc.lua file out of the source code to edit it and then be able to use it as your OSC that MPV uses by allowing option to turn off the compiled osc `osc=no`
Would the same concept not be possible to add in a similar way for the default OSD used in MPV?

Thank You

**Reply by guidocella** [reply]
That's just a consequence of the OSC being written in Lua, which the OSD is not. You need to modify the OSD C code or add options to it.

**Comment by advice2020** [comment]
I put this in the "Ideas" section with the idea of maybe this would be something that would be added in the future, as there is no way for users to make that many modifications to the default OSDs currently.
I am not sure how much coding would be involved in doing so, if this would even be considered? and there do not appear to be any developers that come into the Discussions Ideas section that I am aware of so not sure.
I just figured this would be an all in one solution for editing default OSD items.
The alternative would be if developers would add the ability to make modifications through settings/options but then many separate settings would need to be added to do so.
